osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -116,6 +123,12 @@ protected override bool OnStart()
                 Mods.Value = mods;
             }
 
+            // Default to local leaderboard if the currently selected leaderboard doesn't have scores or is unavailable
+            // perhaps because it requires sign in, requires the beatmap to be ranked, etc.
+            // Also default to local if the details tab is selected so the leaderboard with the new score is visible after returning to song select.
+            if (playBeatmapDetailArea.Leaderboard.State != LeaderboardState.Success || beatmapDetailTab.Value == PlayBeatmapDetailArea.TabType.Details)
+                beatmapDetailTab.Value = PlayBeatmapDetailArea.TabType.Local;
```

Comment:
This is going to overwrite the user setting, ie. change their tab permanently. Is that the intention? I think this could be a bit weird if a user is going down their list of songs and one doesn't have global leaderboards, therefore triggers the tab to change (forever).

Reply:
Yes, that is the intention. Doesn't this only trigger when a beatmap is started instead of the list just being scrolled? I tested a bit locally with friend leaderboard selected (none of my beatmaps have this available) and just by scrolling through my list the selected tab didn't change.

Reply:
You will find most players use down or right arrow keys to listen and view previews of each map in their carousel.

Reply:
I didn't know. I usually just scroll with my tablet and don't really touch my keyboard unless I'm typing.

I still don't quite understand what this has to do with this though because whichever way you navigate through your beatmaps in song select, unless you don't actually start a map this code is never going to be executed because this is in `OnStart`.
I tested it locally and pressing the arrow keys to navigate through the beatmaps never changes the leaderboard to the local one if the currently selected one is unavailable.

Reply:
Hmm, doing it on start is pretty weird. I'd argue that users will be quite confused by this. Will have to think about it a bit.